File: src/main/java/com/manicure/service/AddressService.java
package com.manicure.service;

import java.util.List;

import com.manicure.entity.TbAddress;
import com.manicure.entity.TbAreas;
import com.manicure.entity.TbCities;
import com.manicure.entity.TbProvinces;

public interface AddressService {

	/**
	 * 返回所有省份
	 * @return
	 */
	public List<TbProvinces> findAll();
	
	/**
	 * 根据省id返回城市列表
	 * @param id
	 * @return
	 */
	public List<TbCities> findByProvincesId(String id);
	
	
	/**
	 * 返回地区列表
	 * @param id
	 * @return
	 */
	public List<TbAreas> findByCitiesId(String id);


	/**
	 * 添加地址
	 * @param address
	 */
	public void add(TbAddress address,String userName);

	/**
	 * 修改地址
	 * @param address
	 */
	public void update(TbAddress address);

	/**
	 * 删除
	 * @param id
	 */
	public void delete(Long id);

	/**
	 * 根据用户名查询所有地址
	 * @return
	 */
	public List<TbAddress> findAllAddress(String userName);

	/**
	 * 
	 * @param id
	 * @return
	 */
	public TbAddress findById(Long id);

	/**
	 * 设置默认
	 * @param address
	 */
	public void isDefault(TbAddress address,String userName);

	
}

File: src/main/webapp/js/service/addressService.js
//服务层
app.service('addressService',function($http){
	 //查找所有的省
	this.findAll = function(){
		return $http.get("../address/findAll.do");
	}
	//查询省市
	this.findByProvenceId = function(parentId){
		return $http.get("../address/findByProvenceId.do?parentId="+parentId);
	}
	//查询地区
	this.findByCityId = function(parentId){
		return $http.get("../address/findByCityId.do?parentId="+parentId);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../address/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../address/update.do',entity );
	}
	//列举地址
	this.findAllAddress=function(entity){
		return  $http.get('../address/findAllAddress.do');
	}
	//删除
	this.dele=function(id){
		return  $http.get('../address/delete.do?id='+id);
	}
	
	this.findById=function(id){
		return  $http.get('../address/findById.do?id='+id);
	}
	
	//设置默认
	this.isDefault=function(entity){
		return  $http.post('../address/isDefault.do',entity);
	}
});
